src/processTargets/modifiers/scopeHandlers/TokenScopeHandler.ts

```diff
@@ -0,0 +1,79 @@
+import { Position, Range, TextEditor } from "vscode";
+import { getMatcher } from "../../../core/tokenizer";
+import { Target } from "../../../typings/target.types";
+import { getTokensInRange } from "../../../util/getTokensInRange";
+import { expandToFullLine } from "../../../util/rangeUtils";
+import { TokenTarget } from "../../targets";
+import { ContainedIndices, ScopeHandler } from "./scopeHandler.types";
+
+export default class TokenScopeHandler extends ScopeHandler {
+  protected getEveryTarget(
+    editor: TextEditor,
+    contentRange: Range,
+    isReversed: boolean,
+    hasExplicitRange: boolean
+  ): Target[] {
+    const tokenRanges = getTokensInRange(
+      editor,
+      expandToFullLine(editor, contentRange)
+    ).map(({ range }) => range);
+
+    const filteredTokenRanges = hasExplicitRange
+      ? this.filterRangesByIterationScope(contentRange, tokenRanges)
+      : tokenRanges;
+
+    return filteredTokenRanges.map(
+      (contentRange) =>
+        new TokenTarget({
+          editor,
+          isReversed,
+          contentRange,
+        })
+    );
+  }
+
+  protected getContainingIndicesForPosition(
```

Comment:
Notice how token overwrites the base class behavior for containing scope for a position(empty range)